src/trayicon.rs

```diff
@@ -30,9 +32,23 @@ impl TrayIcon {
 
     pub fn register(&mut self, hwnd: HWND) -> Result<()> {
         self.data.hWnd = hwnd;
-        unsafe { Shell_NotifyIconW(NIM_ADD, &self.data) }
-            .ok()
-            .map_err(|e| anyhow!("Fail to add trayicon, {}", e))
+        unsafe {
+            let result = Shell_NotifyIconW(NIM_ADD, &self.data);
+            if !result.as_bool() {
+                let err = GetLastError();
+                // fix issue #111
+                if err.0 as i32 == E_FAIL.0 {
+                    sleep(Duration::from_secs(2));
+                    Shell_NotifyIconW(NIM_ADD, &self.data)
+                } else {
+                    result
+                }
+            } else {
+                result
+            }
+        }
+        .ok()
```

Comment:
Why do you think a delay of 2 seconds is enough? Shouldn't it be repeated several times?

If `Shell_NotifyIconW` works in thread, after the first failure, it is better to open another thread and try again in it. It should not block the main thread, and the lack of trayicon does not affect its use.

`ok()` will call getLastError internally, and you have called it yourself. Won't this cause problems?
Recommand to only use `ok()` to obtain the execution result and do not call `GetLastError` manually.

Reply:
1. According to the #111 discussion, if the `tray window` of the device is not ready after 2 secs, then it is certain that the device is too behind.
2. I don’t think anyone will switch windows within 2-5 secs after booting or login. so There is no need to process it asynchronously.
3. Calling GetLastError multiple times in the same thread will always get the same value until a new error occurs.

The only thing worth discussing is the number of secs of delay, it could be one of 2,3,4,5, but should not be more than 5.
We don't have to consider it works on devices that should be eliminated.